src/core/utility/file/filePath/FilePath.h

```diff
@@ -366,11 +364,9 @@ class FilePath final {
   bool operator<(const FilePath& other) const;
 
 private:
-  std::unique_ptr<boost::filesystem::path> m_path;
+  std::unique_ptr<fs::path> m_path;
 
-  mutable bool m_exists;
-  mutable bool m_checkedExists;
-  mutable bool m_isDirectory;
-  mutable bool m_checkedIsDirectory;
-  mutable bool m_canonicalized;
+  mutable std::optional<bool> m_exists;
+  mutable std::optional<bool> m_isDirectory;
```

Comment:
Did you check if we need mutable in the code?

Reply:
I will modify code to get rid of mutable keyword.

Reply:
Yes.
As if we removed it, we will need to modify all function that modify m_exists & m_isDirectory which will leads to a lot of changes in Filesystem code.